File: src/main/java/kr/co/infob/common/database/entity/MenuInfo.java
package kr.co.infob.common.database.entity;

import java.util.Date;

import lombok.Getter;
import lombok.Setter;

@Getter @Setter
public class MenuInfo {

	private Long menuSn;          // 메뉴id

	private Long upperMenuSn;     // 부모메뉴id

	private Long resrcSn;         // 리소스일련번호

	private String menuNm;        // 메뉴명

	private String menuUrl;       // 메뉴url

	private Integer menuLevel;    // 메뉴레벨

	private String popupYn;       // 팝업창여부

	private String useYn;         // 사용여부

	private String outptYn;       // 보여주기여부

	private Integer menuOrd;      // 순번

	private String registerId;    // 등록자아이디

	private Date registDttm;      // 등록일시

	private String updusrId;      // 수정자아이디

	private Date updtDttm;        // 수정일시

}

File: src/main/frontend/components/entity/MenuInfo.ts
export interface menuInfo {
    menuSn: number;
    upperMenuSn?: number | null;
    resrcSn: number | null;
    menuNm: string;
    menuUrl: string | null;
    menuLevel: number;
    popupYn: string;
    useYn: string;
    outptYn?: string | null;
    menuOrd: number;
    registerId: string | null;
    registDttm: Date | null;
    updusrId: string | null;
    updtDttm?: Date | null;
}


export interface menuRole {
    menuSn: number;
    roleCd: string | null;
    roleNm: string | null;
    readAuthYn: string | null;
    writeAuthYn: string | null;
}

export interface userMenu extends menuInfo{
    parent?: userMenu | null;
    subMenus?: userMenu[] | null;
    menuRols?: menuRole[] | null;
    realUrl: string;
}

export interface responseUserMenu {
    topMenu: userMenu | null;
    currentMenu: userMenu | null;
    userMenus: userMenu[];
}